sopel/config/core_section.py

```diff
@@ -111,120 +139,271 @@ class CoreSection(StaticSection):
+    """The host for Sopel's database.
 
-        https://docs.sqlalchemy.org/en/latest/core/engines.html
+    Ignored when using SQLite.
     """
 
-    db_user = ValidatedAttribute('db_user')
-    """The user for Sopel's database."""
+    db_name = ValidatedAttribute('db_name')
+    """The name of Sopel's database.
+
+    Ignored when using SQLite.
+    """
 
     db_pass = ValidatedAttribute('db_pass')
-    """The password for Sopel's database."""
+    """The password for Sopel's database.
 
-    db_host = ValidatedAttribute('db_host')
-    """The host for Sopel's database."""
+    Ignored when using SQLite.
+    """
 
     db_port = ValidatedAttribute('db_port')
-    """The port for Sopel's database."""
+    """The port for Sopel's database.
 
-    db_name = ValidatedAttribute('db_name')
-    """The name of Sopel's database."""
+    Ignored when using SQLite.
+    """
+
+    db_type = ChoiceAttribute('db_type', choices=[
+        'sqlite', 'mysql', 'postgres', 'mssql', 'oracle', 'firebird', 'sybase'], default='sqlite')
+    """The type of database Sopel should connect to.
+
+    :default: ``sqlite`` (part of Python's standard library)
+
+    The full list of values Sopel recognizes is:
+
+    * ``firebird``
+    * ``mssql``
+    * ``mysql``
+    * ``oracle``
+    * ``postgres``
+    * ``sqlite``
+    * ``sybase``
+
+    Here are the additional PyPI packages you may need to install to use one of
+    the most commonly requested alternatives:
+
+    mysql
+      ``pip install mysql-python`` (Python 2)
+
+      ``pip install mysqlclient`` (Python 3)
+    postgres
+      ``pip install psycopg2``
+    mssql
+      ``pip install pymssql``
+
+    .. seealso::
+
+        Refer to :ref:`SQLAlchemy's documentation <dialect_toplevel>` for more
+        information about the different dialects it supports.
+
+    """
+
+    db_user = ValidatedAttribute('db_user')
+    """The user for Sopel's database.
+
+    Ignored when using SQLite.
+    """
 
     default_time_format = ValidatedAttribute('default_time_format',
                                              default='%Y-%m-%d - %T%Z')
-    """The default format to use for time in messages."""
+    """The default format to use for time in messages.
+
+    :default: ``%Y-%m-%d - %T%Z``
+
+    Used when plugins format times with :func:`sopel.tools.time.format_time`.
+
+    .. seealso::
+
+        Time format reference is available in the documentation for Python's
+        :func:`time.strftime` function.
+
+    """
 
     default_timezone = ValidatedAttribute('default_timezone', default='UTC')
-    """The default timezone to use for time in messages."""
+    """The default timezone to use for time in messages.
+
+    :default: ``UTC``
+
+    Used when plugins format times with :func:`sopel.tools.time.format_time`.
+    """
 
     enable = ListAttribute('enable')
-    """A whitelist of the only plugins you want to enable."""
+    """A whitelist of the only plugins you want to enable.
```

Comment:
It's not a whitelist, it's a simple list. A whitelist (or, as I prefer, a "safelist") is a list of allowed. In our case, it's a list of enabled and/or activated. Something can be allowed but not enabled, and something can be enabled but not allowed.

If you follow what I mean.

Reply:
Obviously you can see I didn't write this line myself. 😁 I'll `s/white//` it when I'm applying whatever you say to change (if anything) on the `<dl>` nitpick above.